WordPress/src/main/java/org/wordpress/android/datasets/NotificationsTable.java

```diff
@@ -0,0 +1,139 @@
+        return WordPress.wpDB.getDatabase();
+    }
+
+    public static void createTables(SQLiteDatabase db) {
+        db.execSQL("CREATE TABLE IF NOT EXISTS " + NOTIFICATIONS_TABLE + " ("
+                + "id                       INTEGER PRIMARY KEY DEFAULT 0,"
+                + "note_id                  TEXT,"
+                + "type                     TEXT,"
+                + "raw_note_data            TEXT,"
+                + "timestamp                INTEGER," +
+                " UNIQUE (note_id) ON CONFLICT REPLACE"
+                + ")");
+    }
+
+    private static void dropTables(SQLiteDatabase db) {
+        db.execSQL("DROP TABLE IF EXISTS " + NOTIFICATIONS_TABLE);
+    }
+
+    public static ArrayList<Note> getLatestNotes() {
+        return getLatestNotes(20);
+    }
+
+    public static ArrayList<Note> getLatestNotes(int limit) {
+        Cursor cursor = getDb().query(NOTIFICATIONS_TABLE, new String[] {"note_id", "raw_note_data"},
+                null, null, null, null, "timestamp DESC", "" + limit);
+        ArrayList<Note> notes = new ArrayList<Note>();
+        while (cursor.moveToNext()) {
+            String note_id = cursor.getString(0);
+            String raw_note_data = cursor.getString(1);
+            try {
+                Note note = new Note(note_id, new JSONObject(raw_note_data));
+                notes.add(note);
+            } catch (JSONException e) {
+                AppLog.e(AppLog.T.DB, "Can't parse notification with note_id:" + note_id + ", exception:" + e);
+            }
+        }
+        cursor.close();
+        return notes;
+    }
+
+    public static boolean putNote(Note note) {
+        ContentValues values = new ContentValues();
+        values.put("type", note.getType());
+        values.put("timestamp", note.getTimestamp());
+        values.put("raw_note_data", note.getJSON().toString()); // easiest way to store schema-less data
+        values.put("note_id", note.getId());
+
+        long result = getDb().insertWithOnConflict(NOTIFICATIONS_TABLE, null, values, SQLiteDatabase.CONFLICT_REPLACE);
+
+        if (result == -1) {
+            AppLog.e(AppLog.T.DB, "An error occurred while saving the note into the DB -  note_id:" + note.getId());
+        }
+
+        return result != -1;
+    }
+
+    public static void saveNotes(List<Note> notes, boolean clearBeforeSaving) {
+        getDb().beginTransaction();
+        try {
+            if (clearBeforeSaving) {
+                clearNotes();
+            }
+
+            for (Note note: notes) {
+                putNote(note);
+            }
+
+            getDb().setTransactionSuccessful();
+        } finally {
+            getDb().endTransaction();
+        }
+    }
+
+    public static boolean saveNote(Note note, boolean shouldUpdate) {
+        getDb().beginTransaction();
+        boolean saved = false;
+        try {
+             if (!shouldUpdate && NotificationsTable.getNoteById(note.getId()) != null) {
```

Comment:
ok so nitpicking here: it might be better to just querying for the existing of such ID in the database, rather than obtaining, reading and inflating a Note object just to peek at its ID